runExampleRaces throws IllegalStateException

MacAir-2797:entity-services-examples rgupta$ ./gradlew runExampleRaces
:entity-services-examples:compileJava UP-TO-DATE
:entity-services-examples:processResources UP-TO-DATE
:entity-services-examples:classes UP-TO-DATE
:entity-services-examples:runExampleRaces
10:49:33,494 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
10:49:33,495 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
10:49:33,495 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [file:/Users/rgupta/git/entity-services/entity-services-examples/build/resources/main/logback.xml]
10:49:33,619 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
10:49:33,623 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
10:49:33,631 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
10:49:33,686 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [ch.qos.logback] to WARN
10:49:33,687 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [ch.qos.logback] to false
10:49:33,687 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ch.qos.logback]
10:49:33,689 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.marklogic.client] to WARN
10:49:33,689 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [com.marklogic.client] to false
10:49:33,689 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[com.marklogic.client]
10:49:33,689 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.marklogic.semantics] to DEBUG
10:49:33,689 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [com.marklogic.semantics] to false
10:49:33,689 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[com.marklogic.semantics]
10:49:33,689 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.http] to INFO
10:49:33,690 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [org.apache.http] to false
10:49:33,690 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[org.apache.http]
10:49:33,690 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.marklogic.entityservices] to INFO
10:49:33,690 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [com.marklogic.entityservices] to false
10:49:33,690 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[com.marklogic.entityservices]
10:49:33,690 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
10:49:33,691 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
10:49:33,691 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
10:49:33,693 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@46f5f779 - Registering current configuration as safe fallback point
10:49:35.053 [main] INFO  c.m.e.e.ExamplesBase - JSON job started.
10:49:35.562 [main] INFO  c.m.e.e.ExamplesBase - Reading subdirectory models
10:49:35.601 [Thread-1] INFO  c.m.e.e.ExamplesBase - JSON job started.
10:49:35.601 [main] INFO  c.m.e.e.ExamplesBase - JSON job started.
10:49:35.601 [Thread-1] INFO  c.m.e.e.ExamplesBase - Reading subdirectory models
10:49:35.601 [main] INFO  c.m.e.e.ExamplesBase - Reading subdirectory race-data
10:49:35.602 [main] INFO  c.m.e.e.ExamplesBase - Reading subdirectory races
10:49:35.603 [main] INFO  c.m.e.e.ExamplesBase - Reading subdirectory runners
10:49:35.611 [main] INFO  c.m.e.e.ExamplesBase - Reading subdirectory runs
10:49:35.784 [Thread-2] INFO  c.m.e.e.ExamplesBase - JSON job started.
10:49:35.785 [Thread-2] INFO  c.m.e.e.ExamplesBase - Reading subdirectory race-data
10:49:35.785 [Thread-2] INFO  c.m.e.e.ExamplesBase - Reading subdirectory races
10:49:35.786 [Thread-2] INFO  c.m.e.e.ExamplesBase - Reading subdirectory runners
10:49:35.787 [main] INFO  c.m.e.e.ExamplesBase - RDF Load Job started
Exception in thread "main" java.lang.IllegalStateException: Configuration cannot be changed after starting this job or calling add or addAs
10:49:35.791 [Thread-2] INFO  c.m.e.e.ExamplesBase - Reading subdirectory runs
        at com.marklogic.client.datamovement.impl.WriteBatcherImpl.requireNotInitialized(WriteBatcherImpl.java:319)
        at com.marklogic.client.datamovement.impl.WriteBatcherImpl.withTransform(WriteBatcherImpl.java:690)
        at com.marklogic.entityservices.examples.ExamplesBase.importRDF(ExamplesBase.java:138)
        at com.marklogic.entityservices.examples.race.AsIsLoader.lambda$rdfLoad$11(AsIsLoader.java:59)
        at com.marklogic.entityservices.examples.race.AsIsLoader$$Lambda$8/1759482496.run(Unknown Source)
        at com.marklogic.entityservices.examples.race.AsIsLoader.rdfLoad(AsIsLoader.java:61)
        at com.marklogic.entityservices.examples.race.AsIsLoader.loadAsIs(AsIsLoader.java:70)
        at com.marklogic.entityservices.examples.race.ExamplesLoader.main(ExamplesLoader.java:35)
10:49:37.605 [pool-2-thread-1] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:37.609 [pool-1-thread-1] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:37.903 [pool-4-thread-2] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:37.912 [pool-3-thread-4] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:37.914 [pool-3-thread-3] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:37.935 [pool-3-thread-5] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:37.944 [pool-3-thread-1] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:38.306 [pool-4-thread-3] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:38.502 [pool-4-thread-4] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:38.843 [pool-3-thread-2] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:38.974 [pool-3-thread-3] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:38.990 [pool-3-thread-1] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:38.994 [pool-3-thread-4] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.230 [pool-3-thread-5] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.252 [pool-4-thread-5] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.285 [pool-4-thread-3] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.348 [pool-4-thread-2] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.527 [pool-4-thread-4] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.541 [pool-3-thread-1] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.740 [pool-3-thread-2] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.770 [pool-4-thread-1] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.808 [pool-3-thread-3] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.841 [pool-4-thread-4] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:39.867 [pool-4-thread-3] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:40.027 [pool-4-thread-2] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
10:49:40.168 [pool-4-thread-5] INFO  c.m.e.e.ExamplesBase - Loaded batch of documents
:entity-services-examples:runExampleRaces FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':entity-services-examples:runExampleRaces'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_51.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1 mins 8.869 secs
MacAir-2797:entity-services-examples rgupta$

--- a/entity-services-examples/src/main/java/com/marklogic/entityservices/examples/ExamplesBase.java
+++ b/entity-services-examples/src/main/java/com/marklogic/entityservices/examples/ExamplesBase.java
@@ -94,8 +94,6 @@ public abstract class ExamplesBase {
                     // throwable.printStackTrace();
                 });
 
-        moveMgr.startJob(batcher);
-
         return batcher;
     }
 
@@ -139,6 +137,7 @@ public abstract class ExamplesBase {
 
         importOrDescend(referenceDataDir, batcher, collection, Format.TEXT);
 
+        moveMgr.startJob(batcher);
         batcher.flushAsync();
     }
 
@@ -153,6 +152,7 @@ public abstract class ExamplesBase {
         WriteBatcher batcher = newBatcher();
         importOrDescend(jsonDirectory, batcher, toCollection, Format.JSON);
 
+        moveMgr.startJob(batcher);
         batcher.flushAsync();
     }
 
@@ -164,6 +164,7 @@ public abstract class ExamplesBase {
 
         importOrDescend(xmlDirectory, batcher, toCollection, Format.XML);
 
+        moveMgr.startJob(batcher);
         batcher.flushAsync();
     }
 

--- a/entity-services-examples/src/main/java/com/marklogic/entityservices/examples/person/PersonExplorer.java
+++ b/entity-services-examples/src/main/java/com/marklogic/entityservices/examples/person/PersonExplorer.java
@@ -151,7 +151,7 @@ public class PersonExplorer extends ExamplesBase {
                 });
 
         JobTicket ticket = moveMgr.startJob(queryBatcher);
-        queryBatcher.awaitTermination(Long.MAX_VALUE, TimeUnit.DAYS);
+        queryBatcher.awaitCompletion();
         moveMgr.stopJob(ticket);
     }
 

--- a/entity-services-examples/src/main/java/com/marklogic/entityservices/examples/race/Harmonizer.java
+++ b/entity-services-examples/src/main/java/com/marklogic/entityservices/examples/race/Harmonizer.java
@@ -55,7 +55,7 @@ public class Harmonizer extends ExamplesBase {
                 });
 
         JobTicket ticket = moveMgr.startJob(queryBatcher);
-        queryBatcher.awaitTermination(Long.MAX_VALUE, TimeUnit.DAYS);
+        queryBatcher.awaitCompletion();
         moveMgr.stopJob(ticket);
     }
 
@@ -82,7 +82,7 @@ public class Harmonizer extends ExamplesBase {
                 });
 
         JobTicket ticket = moveMgr.startJob(queryBatcher);
-        queryBatcher.awaitTermination(Long.MAX_VALUE, TimeUnit.DAYS);
+        queryBatcher.awaitCompletion();
         moveMgr.stopJob(ticket);
     }
 

